taichi/ir/statements.cpp

```diff
@@ -26,9 +26,15 @@ bool UnaryOpStmt::same_operation(UnaryOpStmt *o) const {
   return false;
 }
 
-ExternalPtrStmt::ExternalPtrStmt(const LaneAttribute<Stmt *> &base_ptrs,
-                                 const std::vector<Stmt *> &indices)
-    : base_ptrs(base_ptrs), indices(indices) {
+ExternalPtrStmt::ExternalPtrStmt(
+    const LaneAttribute<Stmt *> &base_ptrs,
+    const std::vector<Stmt *> &indices,
+    const std::vector<int> &element_shape = std::vector<int>(),
```

Comment:
Can we separate this into two constructors? ;) Specifically I'm interested in if there's any kernels left only proving element_dim without element_shape (I guess yes, those predefined kernels like ext_arr_to_numpy might be still use them. This is totally fine but it's better that we can get it when we read code.

Reply:
Two overloaded constructors were placed at the very beginning. However, I could only find one single usage in the codebase so I merged the constructors. The element_dim is only available in ExternalTensorExpression previously, so they are currently used together for ExternalPtrStmt.

Reply:
I guess my concern here is moving defaults to header files ;)

Reply:
I see :)